components/script/dom/windowproxy.rs

```diff
@@ -516,6 +517,7 @@ impl WindowProxy {
                 referrer_policy,
                 Some(secure),
                 Some(target_document.insecure_requests_policy()),
+                target_document.has_trustworthy_ancestor_origin(), // sebsebmc: is this context non-nested?
```

Comment:
It can be non-nested, but it can also be nested given a target like `_parent`.

Reply:
I think this should be more correct now, though the fact that the secure context and insecure requests policy seem to be passed on without inspecting this makes me wonder.

Reply:
I'm going to file a followup to review this more closely (including the insecure requests policy and secure context) since this is a good point.